### ISSUE TITLE
Error when running ./vendor/bin/psalm --init

### ISSUE THREAD

**Issue by d3ce1t** [vimeo/psalm#4755]
I've just installed psalm following installation instructions. But I can't continue because `./vendor/bin/psalm --init` gives a fatal error.

The error seems to be similar to the one found by Decomposer in google/apiclient as you can see in [#1256](https://github.com/googleapis/google-api-php-client/issues/1256)

```console
devel:myproject$ ./vendor/bin/psalm --init
Calculating best config level based on project files
Calculating best config level based on project files
Scanning files...
PHP Fatal error:  Interface 'GuzzleHttp\Event\SubscriberInterface' not found in /app/vendor/google/auth/src/Subscriber/AuthTokenSubscriber.php on line 37
Fatal error: Interface 'GuzzleHttp\Event\SubscriberInterface' not found in /app/vendor/google/auth/src/Subscriber/AuthTokenSubscriber.php on line 37
```

Is there any workaround so that vendor directory can be ignored?

**Comment by orklah** [comment]
Sounds like a dependancy issue. Do you have an exotic composer installation or outdated packages that could cause the issue?

Apart from that, you could probably create your psalm.xml by yourself, but there are big chances that you'll encounter this issue once you run the analysis

**Comment by muglug** [comment]
You can skip this inference by providing a directory like

```
vendor/bin/psalm --init src
```

Using the `--debug` flag would confirm this, but I suspect that Psalm is unable to figure out where a class lives via composer. If it fails there, it falls back to PHP's Reflection API. That reflection API will try to load the class, and in this case that loading causes a fatal error due to a missing dependency.

**Comment by d3ce1t** [comment]
> Sounds like a dependancy issue. Do you have an exotic composer installation or outdated packages that could cause the issue?

Looks like I was pinned to an old google/apiclient-services (v118). Updating it to the last version (v156) made the trick.

Now psalm --init works as expected

Thank you!

**Comment by albanh** [comment]
I'm using the v156 og google/apiclient-services but still receive the same error when running` ./vendor/bin/psalm --init`

**Comment by boyerdalton** [comment]
I'm also using v156 and receive this same error

**Comment by Daniel-Griffiths** [comment]
Same issue here

**Comment by d3ce1t** [comment]
**TLDR** You will hit this problem when you have `google/auth` and `guzzlehttp/guzzle` > 5 as dependencies. Downgrading to Guzzle 5 should work. But I wouldn't recommend because Guzzle 5 reached end-of-life.

I'm revisiting this again after having seen that you are facing the same problem I had. I've tried to init psalm again (first I've removed my `psalm.xml` file) to check the fix I proposed was still valid. It wasn't. Somehow it worked for me in the past and I thought the only change I made was the update of `google/apiclient-services`. But likely I did something else. My apologies for the confusion.

After examining this issue again I've found where the exact problem is. The library `google/auth` (a dependency of `google/apiclient`) has a file at [google/auth/src/Subscriber/AuthTokenSubscriber.php](https://github.com/googleapis/google-auth-library-php/blob/master/src/Subscriber/AuthTokenSubscriber.php) using `GuzzleHttp\Event\SubscriberInterface`. However, `GuzzleHttp\Event\SubscriberInterface` only exists in Guzzle 5 but not in 6 and 7. In other words, you will hit this problem when you have google/auth and guzzle > 5.

Likely what worked for me was downgrading Guzzle to version 5.

### UPDATE
This is the content of the `psalm.xml` file that I got created after executing `psalm --init ` time ago:

```xml
<?xml version="1.0"?>
<psalm
    errorLevel="8"
    resolveFromConfigFile="true"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
>
    <projectFiles>
        <directory name="app" />
        <directory name="lib" />
        <ignoreFiles>
            <directory name="vendor" />
        </ignoreFiles>
    </projectFiles>
</psalm>
```
Modify projectFiles section to fit your project organization and place it to your project's root directory. After that, you should be able to execute `./vendor/bin/psalm` to analyse your code.

**Comment by d3ce1t** [comment]
I'm reopening this issue to make a suggestion. I haven't read the code so take it with a pinch of salt.

IMHO I think vendor directory should be skipped during initialization. After all, the generated psalm.xml file will have it ignored anyway. CMIIW this should have avoided this issue completely and improve user experience of newcomers like me.
